updatable_timer/workflow.py

```diff
@@ -0,0 +1,35 @@
+from datetime import datetime, timezone
+from typing import Optional
+
+from temporalio import workflow
+
+from updatable_timer.updatable_timer_lib import UpdatableTimer
+
+
+@workflow.defn
+class Workflow:
+    def __init__(self):
+        self.timer: Optional[UpdatableTimer] = None
+
+    @workflow.run
+    async def run(self, wake_up_time: float):
+        self.timer = UpdatableTimer(
+            datetime.fromtimestamp(wake_up_time, tz=timezone.utc)
+        )
+        await self.timer.sleep()
+
+    @workflow.signal
+    async def update_wake_up_time(self, wake_up_time: float):
+        # Deals with situation when the signal method is called before the run method.
+        # This happens when a workflow task is executed after a signal is received
+        # or when a workflow is started using the signal-with-start.
+        await workflow.wait_condition(lambda: self.timer is not None)
+        assert self.timer is not None  # for mypy
```

Comment:
We have a neat, fairly new feature for `@workflow.init` that lets you operate on workflow input on the constructor, so can change this to:

```suggestion
@workflow.defn
class Workflow:
    @workflow.init
    def __init__(self, wake_up_time: float) -> None:
        self.timer = UpdatableTimer(
            datetime.fromtimestamp(wake_up_time, tz=timezone.utc)
        )

    @workflow.run
    async def run(self, wake_up_time: float) -> None:
        await self.timer.sleep()

    @workflow.signal
    async def update_wake_up_time(self, wake_up_time: float):
```

(also can remove the `assert self.timer is not None  # for mypy` in the query)

Reply:
Nice! So much clearer!